zerver/webhooks/gogs/view.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# vim:fenc=utf-8
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.models import Client, UserProfile
+from zerver.lib.webhooks.git import get_push_commits_event_message, \
+    get_pull_request_event_message, SUBJECT_WITH_BRANCH_TEMPLATE, \
+    SUBJECT_WITH_PR_OR_ISSUE_INFO_TEMPLATE
+
+from django.http import HttpRequest, HttpResponse
+from typing import Dict, Any, Iterable, Optional, Text
+
+
+def format_push_event(payload):
+    # type: (Dict[str, Any]) -> Text
+
+    for commit in payload['commits']:
+        commit['sha'] = commit['id']
+
+    data = {
+        'user_name': payload['sender']['username'],
+        'compare_url': payload['compare_url'],
+        'branch_name': payload['ref'].replace('refs/heads/', ''),
+        'commits_data': payload['commits']
+    }
+
+    return get_push_commits_event_message(**data)
+
+
+def format_new_branch_event(payload):
+    # type: (Dict[str, Any]) -> Text
+
+    body_template = '\n{username} created [{branch}]({html_url}/src/{branch}) branch'
+    data = {
+        'username': payload['sender']['username'],
+        'html_url': payload['repository']['html_url'],
+        'branch': payload['ref']
+    }
+    return body_template.format(**data)
+
+
+def format_pull_request_event(payload):
+    # type: (Dict[str, Any]) -> Text
+
+    data = {
+        'user_name': payload['pull_request']['user']['username'],
+        'action': payload['action'],
+        'url': payload['pull_request']['html_url'],
+        'number': payload['pull_request']['number'],
+        'target_branch': payload['pull_request']['head_branch'],
+        'base_branch': payload['pull_request']['base_branch'],
+    }
+
+    if payload['pull_request']['merged']:
+        data['user_name'] = payload['pull_request']['merged_by']['username']
+        data['action'] = 'merged'
+
+    return get_pull_request_event_message(**data)
+
+
+@api_key_only_webhook_view('Gogs')
+@has_request_variables
+def api_gogs_webhook(request, user_profile, client,
+                     payload=REQ(argument_type='body'),
+                     stream=REQ(default='commits')):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Any], Text) -> HttpResponse
+
+    body = ''
+    event = request.META['HTTP_X_GOGS_EVENT']
+
+    try:
+        if event == 'push':
+            topic = SUBJECT_WITH_BRANCH_TEMPLATE.format(
+                repo = payload['repository']['name'],
```

Comment:
there should be no space around `=`.

correct examples;
`funtction(arg=value)`
`new_variable = 'new value'`